operator/internal/manifests/var.go

```diff
@@ -385,3 +387,30 @@ func podSecurityContext(withSeccompProfile bool) *corev1.PodSecurityContext {
 
 	return &context
 }
+
+func addProxyEnvVar(clusterProxy *lokiv1.ClusterProxy, podSpec corev1.PodSpec) corev1.PodSpec {
```

Comment:
Two things here to improve:

First we need to take care not to append only the env vars, because in the likely event that they exist (e.g. any k8s component adding them), we will add them twice and get such an error:
```
Jan  1 00:00:00 || <no msg> _ts="2022-10-17T09:50:36.932315127Z" _component="loki-operator_KubeAPIWarningLogger" _message="spec.template.spec.containers[0].env[3].name: duplicate name \"NO_PROXY\""
```
This will lead IIRC correct to some audit log, but I cannot recall 100%.

Next we can improve the nesting here by circuit breaking the if statements separately, e.g.:
```go
func addProxyEnvVar(clusterProxy *lokiv1.ClusterProxy, podSpec corev1.PodSpec) corev1.PodSpec {
	if clusterProxy == nil {
		return podSpec
	}

	if clusterProxy.ReadVarsFromEnv {
		for i, container := range podSpec.Containers {
			podSpec.Containers[i].Env = append(container.Env, proxy.ReadProxyVarsFromEnv()...)
		}

		return podSpec
	}

	for i, container := range podSpec.Containers {
		podSpec.Containers[i].Env = append(container.Env,
			corev1.EnvVar{
				Name:  "HTTP_PROXY",
				Value: clusterProxy.HTTPProxy,
			},
			corev1.EnvVar{
				Name:  "HTTPS_PROXY",
				Value: clusterProxy.HTTPSProxy,
			},
			corev1.EnvVar{
				Name:  "NO_PROXY",
				Value: clusterProxy.NoProxy,
			},
			corev1.EnvVar{
				Name:  "NO_PROXY",
				Value: clusterProxy.NoProxy,
			})
	}

	return podSpec
}
```